app/routes/__index/certificate/cert.ts

```diff
@@ -0,0 +1,28 @@
+import type { LoaderArgs } from '@remix-run/node';
+import { Response } from '@remix-run/node';
+import { requireUsername } from '~/session.server';
+import { getCertificateByUsername } from '~/models/certificate.server';
+
+export async function loader({ request }: LoaderArgs) {
+  const username = await requireUsername(request);
+
+  try {
+    const certificate = await getCertificateByUsername(username);
+
+    // If we don't have a certificate, we're not ready to do this yet
+    if (!certificate.certificate) {
+      throw new Response('Certificate Key was not found', { status: 409 });
```

Comment:
I think 404 makes sense too.

Reply:
Isn't 409 for when the request is not in a proper state? In this case the certificate can exists at a pending state, but it doesn't have either privkey or certkey, thus we throw an error? If the certificate doesn't exists it throws a 404

Reply:
You can argue it both ways: the record isn't available (404) and also because it's not ready (409).

Reply:
Oh, I was confused by the comment mentioning that it will happen if a certificate doesn't exist.
I am fine with either but could you update the comment to mention that we will return 409 if the certificate is in a pending state?

Reply:
Oh yeah I get where the confusion might be sure I'll add the comment